src/web/client/test/integration/errorHandler.test.ts

```diff
@@ -0,0 +1,634 @@
+        const _mockShowErrorMessage = stub(vscode.window, "showErrorMessage");
+
+        //Action
+        showErrorDialog(errorString);
+
+        //Assert
+        const options = { detail: undefined, modal: true };
+        assert.calledOnceWithExactly(
+            _mockShowErrorMessage,
+            errorString,
+            options
+        );
+    });
+
+    it("removeEncodingFromParameters_shouldDecodeSCHEMAVERSIONAndWEBSITENAME", () => {
+        //Act
+        const queryParamsMap = new Map<string, string>([
+            [schemaKey.SCHEMA_VERSION, "3A%20test_file_name"],
+            [queryParameters.WEBSITE_NAME, "3A%20test_test_webSiteName"],
+        ]);
+
+        //Action
+        removeEncodingFromParameters(queryParamsMap);
+
+        //Assert
+        const websiteName = queryParamsMap.get(
+            queryParameters.WEBSITE_NAME
+        ) as string;
+
+        const schemaFileName = queryParamsMap.get(
+            schemaKey.SCHEMA_VERSION
+        ) as string;
+
+        expect(schemaFileName).eq("3A test_file_name");
+        expect(websiteName).eq("3A test_test_webSiteName");
+    });
+
+    it("removeEncodingFromParameters_shouldMapPORTALSFOLDERNAMEDEFAULT_whenWebsiteNameIsBlank_", () => {
+        //Act
+        const queryParamsMap = new Map<string, string>([
+            [schemaKey.SCHEMA_VERSION, "3A%20test_file_name"],
+            [queryParameters.WEBSITE_NAME, ""],
+        ]);
+
+        //Action
+        removeEncodingFromParameters(queryParamsMap);
+
+        //Assert
+        const websiteName = queryParamsMap.get(
+            queryParameters.WEBSITE_NAME
+        ) as string;
+
+        const schemaFileName = queryParamsMap.get(
+            schemaKey.SCHEMA_VERSION
+        ) as string;
+
+        expect(schemaFileName).eq("3A test_file_name");
+        expect(websiteName).eq(PORTALS_FOLDER_NAME_DEFAULT);
+    });
+
+    it("checkMandatoryParameters_shouldReturnTrueWhenAppNameIsPortalAndEntityAndEntityIdHavingValues", () => {
+        //Act
+        const appName = "portal";
+        const entity = "entity";
+        const entityId = "3";
+        const queryParamsMap = new Map<string, string>([
+            [queryParameters.ORG_URL, "url"],
+            [queryParameters.DATA_SOURCE, "SQL"],
+            [schemaKey.SCHEMA_VERSION, "1.0.0"],
+            [
+                queryParameters.WEBSITE_ID,
+                "ed9a6c19-5ab6-4f67-8c35-2423cff958c4",
+            ],
+        ]);
+
+        //Action
+        const result = checkMandatoryParameters(
+            appName,
+            entity,
+            entityId,
+            queryParamsMap
+        );
+
+        //Assert
+        expect(result).true;
+    });
+
+    it("checkMandatoryParameters_shouldReturnFalseWhenEntityIsBlank", () => {
+        //Act
+        const appName = "portal";
+        const entity = "";
+        const entityId = "3";
+        const queryParamsMap = new Map<string, string>([
```

Comment:
I see lot of common code in each test case. Extract them and write a single method for each common code

Reply:
Different different values are being served